### PR TITLE
Some improvement for terraform-provider-openstack jobs

### PR DESCRIPTION
**Pull request by h00130372**
1. do not use uuid directly, to get the id quering by name
2. can not create flavor in public clouds, so let the tests fail first

### CODE REVIEW

File: zuul.d/projects.yaml [hunk 1]
```diff
@@ -120,18 +120,18 @@
...
-#      jobs:
-#        - terraform-provider-openstack-acceptance-test-telefonica:
-#            branches: master
-#        - terraform-provider-openstack-acceptance-test-opentelekomcloud:
-#            branches: master
-#        - terraform-provider-openstack-acceptance-test-orange:
-#            branches: master
-#        - terraform-provider-openstack-acceptance-test-huaweicloud:
-#            branches: master
+- project:
```
Comment (kiwik) [inline zuul.d/projects.yaml]: There is an open issue about terraform jobs cleanup https://github.com/theopenlab/openlab/issues/136 , we can address that issue in this PR too.

File: playbooks/terraform-provider-openstack-acceptance-test-public-clouds/run.yaml [hunk 2]
```diff
@@ -2,40 +2,38 @@
...
+          export OS_NETWORK_ID="$(openstack network show $OS_NETWORK_NAME -f value -c id)"
+          export OS_VPC_ID="$(openstack router show openlab-jobs-vpc -f value -c id)"
 
           export OS_SWIFT_ENVIRONMENT=1
 
           # Run except the DNS/FW/LB test 100 testcases at a time
           testcases=`go test ./openstack/ -v -list 'Acc'`
           testcases=`echo "$testcases" | sed '$d' | grep -v -e FW -e LB`
-          echo "$testcases" | xargs -t -n100 sh -c 'TF_LOG=DEBUG TF_ACC=1 go test ./openstack -v -timeout 120m -run $(echo "$@" | tr " " "|")' argv0 2>&1 | tee $TEST_RESULTS_TXT
+          echo "$testcases" | xargs -t -n100 sh -c 'TF_LOG=DEBUG OS_DEBUG=1 TF_ACC=1 go test ./openstack -v -timeout 120m -run $(echo "$@" | tr " " "|")' argv0 2>&1 | tee $TEST_RESULTS_TXT
```
Comment (kiwik) [inline playbooks/terraform-provider-openstack-acceptance-test-public-clouds/run.yaml]: `OS_DEBUG=1` maybe cause log file size exceed zuul job log limit

File: playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  roles:
+    - export-cloud-openrc
+  tasks:
+    - name: Clean up resources for public clouds & terraform-provider-openstack acceptance tests
+      shell:
+        cmd: |
+          set -e
+          set -x
+          # clean up instances
+          for ins_id in `openstack server list -f value -c ID -c Name |grep instance_1 | awk -F ' ' '{print $1}'`
```
Comment (kiwik) [inline playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml]: you can add option `--no-name-lookup` for command `server list`, that is speeder than default option

File: playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          for vol_id in `openstack volume list -f value -c ID -c Name |grep -E  'volume_1|volume_1-updated' | awk -F ' ' '{print $1}'`
+          do
+              openstack volume delete "${vol_id}"
+          done
+
+          # clean up ports
+          for port_id in `openstack port list -f value -c ID -c Name |grep port_1 | awk -F ' ' '{print $1}'`
+          do
+              openstack port delete "${port_id}"
+          done
```
Comment (kiwik) [inline playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml]: some ports belong to `network_1` should be deleted too.

File: playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          set -e
+          set -x
+          # clean up instances
+          for ins_id in `openstack server list -f value -c ID -c Name |grep instance_1 | awk -F ' ' '{print $1}'`
+          do
+              openstack server delete "${ins_id}"
+          done
+
+          # clean up keypairs
+          for kp_name in `openstack keypair list -f value -c Name |grep kp_1 | awk -F ' ' '{print $1}'`
```
Comment (liu-sheng) [inline playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml]: do not need the {print $1} in L18.

and all the `-F ' ' ` is not needed

File: playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          do
+              openstack keypair delete "${kp_name}"
+          done
+
+          # clean up security groups
+          for sg_id in `openstack security group list -f value -c ID -c Name |grep -E 'security_group|security_group_1|security_group_2' | awk -F ' ' '{print $1}'`
```
Comment (liu-sheng) [inline playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml]: `security_group|security_group_1|security_group_2` can be abbreviated as `security_group`

File: playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+          done
+
+          # clean up images
+          for image_id in `openstack image list -f value -c ID -c Name |grep 'Rancher TerraformAccTest' | awk -F ' ' '{print $1}'`
+          do
+              openstack image delete "${image_id}"
+          done
+
+          # clean up volumes
+          for vol_id in `openstack volume list -f value -c ID -c Name |grep -E  'volume_1|volume_1-updated' | awk -F ' ' '{print $1}'`
```
Comment (liu-sheng) [inline playbooks/terraform-provider-openstack-acceptance-test-public-clouds/post.yaml]: `grep -E  'volume_1|volume_1-updated' ` ==  `grep volume_1`

**Pull request merged**